internal/api/v1beta1/app_types.go

```diff
@@ -236,6 +236,19 @@ type AppSpec struct {
 
 	// SecurityContext specifies security settings for a pod/app, which get applied to all containers.
 	SecurityContext *v1.PodSecurityContext `json:"securityContext,omitempty"`
+
+	// VolumeClaimTemplates is a list of an app's volumeClaimTemplates
+	VolumeClaimTemplates []PersistentVolumeClaim `json:"volumeClaimTemplates,omitempty"`
+
+	// Type specifies whether an app should be a deployment or a statefulset
+	Type *string `json:"type,omitempty"`
```

Comment:
maybe
```golang

// AppSpec defines the desired state of App.
type AppSpec struct {

	// Type specifies whether an app should be a deployment or a statefulset
	// +kubebuilder:validation:default:=Deployment
	Type *AppType `json:"type,omitempty"`
}

// +kubebuilder:validation:Enum=Deployment;StatefulSet
type AppType string
const (
	DeploymentType AppType = "Deployment"
	StatefulSetAppType AppType = "StatefulSet"
)

func (spec AppSpec) GetType() AppType {
	if spec.Type == nil {
		return DeploymentType
	}
	return *spec.Type
}
```

"Deployment" and "Statefulset" are used multiple times and having a dedicated enum for them can make life a bit easier

Reply:
Great suggestion! I implemented it and it works very well.